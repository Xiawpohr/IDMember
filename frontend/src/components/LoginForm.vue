<template>
  <form @submit.prevent="$emit('submitted', auth)">
    <input type="text" placeholder="Email" v-model="Email">
    <input type="password" placeholder="Password" v-model="password">
    <button type="submit">Log In</button>
  </form>
</template>

<script>
export default {
  data () {
    return {
      email: '',
      password: ''
    }
  },
  computed: {
    auth() {
      return {
        email: this.email,
        password: this.password
      }
    }
  }
}
</script>
