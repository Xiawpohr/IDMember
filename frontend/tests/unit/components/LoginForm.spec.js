import { shallowMount } from '@vue/test-utils'
import LoginForm from '@/components/LoginForm.vue'

const auth = { email: 'test@example.com', password: '123456' }

let wrapper

beforeEach(() => {
  wrapper = shallowMount(LoginForm)
})

it('should render the component', () => {
  expect(wrapper.html()).toMatchSnapshot()
})

it('should call submittd event when submitting the form', () => {
  wrapper.setData({
    ...auth
  })
  const button = wrapper.find('button')
  button.trigger('click')
  button.trigger('submit')
  expect(wrapper.emitted('submitted')[0]).toEqual([auth])
})
