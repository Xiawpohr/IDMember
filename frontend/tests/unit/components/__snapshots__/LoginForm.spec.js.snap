// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the component 1`] = `<form><input type="text" placeholder="Email"> <input type="password" placeholder="Password"> <button type="submit">Log In</button></form>`;
